tx-pool/src/pool.rs

```diff
@@ -226,58 +199,43 @@ impl TxPool {
 
     /// Returns true if the tx-pool contains a tx with specified id.
     pub fn contains_proposal_id(&self, id: &ProposalShortId) -> bool {
-        self.pending.contains_key(id)
-            || self.proposed.contains_key(id)
-            || self.orphan.contains_key(id)
+        self.pending.contains_key(id) || self.proposed.contains_key(id)
     }
 
     /// Returns tx with cycles corresponding to the id.
     pub fn get_tx_with_cycles(
         &self,
         id: &ProposalShortId,
-    ) -> Option<(TransactionView, Option<Cycle>)> {
+    ) -> Option<(&TransactionView, Option<Cycle>)> {
```

Comment:
I think the `Cycle` cannot be `None` after refactoring, we can change the return type to:

```suggestion
    ) -> Option<(&TransactionView, Cycle)> {
```